### PR TITLE
feat(eslint-plugin): Support TypeScript@4.

### PR DESCRIPTION
**Pull request by doniyor2109**

### CODE REVIEW

File: packages/tsconfig/tsconfig.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     //
 
     "noEmit": true,
+    "incremental": true,
```
Comment (umidbekkarimov) [inline packages/tsconfig/tsconfig.json:8]: what this gives us? also, we will have to add `tsconfig.tsbuildinfo` to the `.gitignore` everywhere after that update.

Comment (doniyor2109) [reply packages/tsconfig/tsconfig.json:8]: It gives great speed. I have checked it on CTMS and it decreased typecheking from 30 sec to 8 sec.

Comment (doniyor2109) [reply packages/tsconfig/tsconfig.json:8]: Or we can make this option turned on for each project

Comment (umidbekkarimov) [reply packages/tsconfig/tsconfig.json:8]: yeah, it's better to enable it manually where we want to.

**Review by umidbekkarimov — APPROVED** [review]

**Pull request merged**